zipline/data/us_equity_pricing.py

```diff
@@ -528,6 +528,16 @@ def _spot_col(self, colname):
     def get_last_traded_dt(self, asset, day):
         volumes = self._spot_col('volume')
         search_day = day
+
+        if search_day >= asset.end_date:
+            # move to the last day before the asset end date
+            try:
+                search_day = self._calendar[
+                    self._calendar.get_loc(asset.end_date) - 1
```

Comment:
Do we want the last entry if `get_loc` raises a KeyError (which would happen if the search day isn't in the calendar)?  Wondering if the try/except block is too broad.